File: cineManagement/cineManagementDatabaseFirst/Models/DTOs/SalaCineDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace cineManagementDatabaseFirst.Models.DTOs
{
    public class SalaCineDTO
    {
        public int SalaId { get; set; }
        public string Nombre { get; set; }
        public int? Estado { get; set; }
        public bool? EsActivo { get; set; }
    }

    public class SalaCineCreateDTO
    {
        [Required]
        public string Nombre { get; set; }
        public int? Estado { get; set; }
    }

    public class SalaCineUpdateDTO
    {
        [Required]
        public int SalaId { get; set; }
        [Required]
        public string Nombre { get; set; }
        public int? Estado { get; set; }
    }
}

File: cineManagementFront/src/app/models/dtos/salaCine.dto.ts
import { PeliculaDTO } from "./pelicula.dto";

export interface SalaCineDTO {
  salaId: number;
  nombre: string;
  estado?: number;
  esActivo?: boolean;
}

export interface SalaCineCreateDTO {
  nombre: string;
  estado?: number;
}

export interface SalaCineUpdateDTO {
  salaId: number;
  nombre: string;
  estado?: number;
}

export interface SalaEstadoDTO {
  salaId: number;
  nombre: string;
  estado?: number;
  cantidadPeliculas: number;
  estadoSala: string;
  peliculas: PeliculaDTO[];
}

export interface SalaCineDeactivateDTO {
  salaId: number;
  esActivo: boolean;
}
